aioredlock/algorithm.py

```diff
@@ -119,7 +137,10 @@ def log(self):
 
             raise
 
-        return Lock(self, resource, lock_identifier, valid=True)
+        lock = Lock(self, resource, lock_identifier, valid=True)
+        self.watchdogs[lock.resource] = asyncio.ensure_future(self.__auto_extend(lock, self.auto_extend_count))
+
+        return lock
 
     async def extend(self, lock):
```

Comment:
Manual call of `extend` may reset watchdog's count to `auto_extend_count`.
It might be done by passing `auto_extend_count` to `extend` like so:
```python
async def extended(self, lock, auto_extend_count=None):
    if auto_extend_count is None:
        auto_extend_count = self.auto_extend_count
````

Reply:
I think it's better we split auto_extend and extend in 2 methods.
If we just use extend method to do both job, it might confuse users.
If we do as your way, user will have no way to extend his lock once he set auto_extend_count to 0